packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -952,6 +965,15 @@ class CachedLocalEngineArtifacts implements LocalEngineArtifacts {
     throw Exception('Unable to find $genSnapshotName');
   }
 
+  String _libFlutterEnginePath() {
+    final String libFlutterEngineName = _artifactToFileName(Artifact.libFlutterEngine)!;
+    final String libFlutterEnginePath = _fileSystem.path.join(engineOutPath, libFlutterEngineName);
+      if (_processManager.canRun(libFlutterEnginePath)) {
```

Comment:
we won't be running this file, right?

Reply:
In practice on my machine _processManager.canRun() effectively just checks if the file exists, [here](https://github.com/google/process.dart/blob/master/lib/src/interface/local_process_manager.dart) ln124 and [here](https://github.com/google/process.dart/blob/master/lib/src/interface/common.dart) ln 53. 

I can't imagine that it will actually run on my machine. it will only be executed on the custom device as far as I understand.

Reply:
This could just use the file APIs to check that it exists.

Reply:
> This could just use the file APIs to check that it exists.

Yeah, something like `if (_fileSystem.file(libFlutterEnginePath).existsSync()) {`